### ISSUE TITLE
BIM: Implement GBXML export

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#5654]
Issue imported from https://tracker.freecad.org/view.php?id=2543

* **Reporter:** yorik
* **Date submitted:** 5/12/2016
* **FreeCAD version:** 
* **Category:** Feature
* **Status:** assigned
* **Tags:** Bountysource, fundable, funded

# Original report text

Implement the possibility to export a BIM model as GBXML

# Additional information

Started in https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Arch/importGBXML.py
Downstream ticket: https://github.com/yorikvanhavre/BIM_Workbench/issues/6
Bountysource: https://www.bountysource.com/issues/65465675-implement-gbxml-export

# Other bug information

* **Priority:** normal
* **Severity:** minor
* **Category:** Feature
* **OS: Debian Testing 64bit**
* **Platform:** PC
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by yorik 2016-10-06 22:45
Some more ideas to implement before actual export:

- Add some general settings to the project (site or building), maybe linked/embedded osm file?
- Add tool to subdivide solid faces with another solid (useful for FEM too)
- Base everything on spaces. What gets exported are spaces.
- Ability to mark faces of spaces as 1) shared 2) exterior 3) ground 4) window
- Ability to define thermal zones and add spaces to these thermal zones (use floors maybe?)
- Ability to retrieve a building's floor area

Also check the .osm format of openstudio. Openstudio itself will allow to set many more settings and do the necessary export to radiance and energyplus.

Basically from FreeCAD we should be able to do the same as what sketchup does, that is:

- create simple geometry (spaces)
- define what is shared wall, what is window, etc (a way to do that easily/automatically?)
- create shadowing devices (not sure how to do that, creates non-solid/manifold objects)
- give these spaces a name, a use/function, a thermal zone.

---

### Comment by Kunda1 2018-10-04 12:54
Added https://github.com/yorikvanhavre/BIM_Workbench/issues/6 to the summary

**Comment by KeithSloan** [comment]
Probably rename but have started work on a Workbench https://github.com/KeithSloan/OpenStudioWorkBench\

The purpose being to create GbXml files from FreeCAD
